src/classlibnative/bcltype/system.cpp

```diff
@@ -99,12 +99,142 @@ FCIMPL0(INT64, SystemNative::__GetSystemTimeAsFileTime)
 FCIMPLEND;
 
 
+#ifndef FEATURE_PAL
+typedef int (WINAPI *NtQuerySystemInformationProc) (int SystemInformationClass, PVOID SystemInformation, ULONG SystemInformationLength, PULONG ReturnLength OPTIONAL);
+typedef struct _SYSTEM_LEAP_SECOND_INFORMATION {
+    BOOLEAN Enabled;
+    ULONG Flags;
+} SYSTEM_LEAP_SECOND_INFORMATION, *PSYSTEM_LEAP_SECOND_INFORMATION;
+
+#define SystemLeapSecondInformation 206
+
+
+BOOL QCALLTYPE SystemNative::IsLeapSecondsSupportedSystem()
+{
+    QCALL_CONTRACT;
+
+    BOOL ret = FALSE;
+
+    BEGIN_QCALL;
+
+    SYSTEM_LEAP_SECOND_INFORMATION slsi;
+    HINSTANCE hNtDll = CLRLoadLibrary(W("ntdll.dll"));
```

Comment:
Can we just PInvoke into `ntdll.lib` all in C# like you have done it for CoreRT version of this change?

Reply:
I was not sure about if there will be any perf difference. corert I didn't have the choice. if you think perf wil not be issue I can convert this to pinvoke.

Reply:
The one-time initialization is certainly not a performance sensitive path.

Reply:
(The rest can stay as FCalls for now. There are probably a few perf sensitive ones.)